### ISSUE TITLE
Keptn 0.6.0.beta EKS installation failed

### ISSUE THREAD

**Issue by checkelmann** [keptn/keptn#1154]
**Environment**

* Client OS (e.g., Linux, OSX, Windows):
* Keptn Version (`keptn version`): CLI version: 0.6.0.beta
* Kubernetes Cloud Provider (e.g., GKE, AKS): EKS 1.14
* Kubernetes version (`kubectl version`):
Server Version: version.Info{Major:"1", Minor:"14+", GitVersion:"v1.14.8-eks-b7174d", GitCommit:"b7174db5ee0e30c94a0b9899c20ac980c0850fc8", GitTreeState:"clean", BuildDate:"2019-10-18T17:56:01Z", GoVersion:"go1.12.10", Compiler:"gc", Platform:"linux/amd64"}

**Affected Component**

* [X] CLI
* [X] Kubernetes Integration
* [X] Istio

**Describe the bug**
Installation on EKS is failing.
CoreDNS Upgrade was made.

```
keptn install --platform=eks
Installing Keptn ...
Please enter the following information or press enter to keep the old value:
Cluster Name [dynatrace-demo]: 
AWS Region [eu-west-1]: 

Please confirm that the provided information is correct: 
Cluster Name: dynatrace-demo
AWS Region: eu-west-1

Is this all correct? (y/n)
y
Using a file-based storage for the key because the password-store seems to be not set up.
Deploying Keptn installer pod ...
Installer pod deployed successfully.
Getting logs of pod installer-6d42z
Testing connection to Kubernetes API
Connection to Kubernetes API successful
Installing Istio on EKS (this might take a while)
Installing Istio failed.
Error occured with message: [keptn|ERROR] [2019-11-22 14:50:11] Installing Istio failed.
Stopping keptn installation. Already created resources are not deleted; run keptn uninstall to clean-up.
Error occured with message: [keptn|ERROR] [2019-11-22 14:50:11] Stopping keptn installation. Already created resources are not deleted; run keptn uninstall to clean-up.
Error: Keptn installation was unsuccessful
```

```
kubectl logs installer-6d42z -f
##########
branch: master
repository: https://github.com/keptn/keptn
commitlink: https://github.com/keptn/keptn/commit/2ee84eb0551b46ab076380541ed4cd2482219085
repolink: https://github.com/keptn/keptn/tree/2ee84eb0551b46ab076380541ed4cd2482219085
travisbuild: https://travis-ci.org/keptn/keptn/jobs/615150387
timestamp: 20191121.1924
##########
[keptn|INFO] [2019-11-22 14:49:24] Testing connection to Kubernetes API
NAME              STATUS   AGE
default           Active   17m
dynatrace         Active   6m22s
kube-node-lease   Active   17m
kube-public       Active   17m
kube-system       Active   17m
[keptn|INFO] [2019-11-22 14:49:24] Connection to Kubernetes API successful
namespace/keptn created
[keptn|INFO] [2019-11-22 14:49:25] Installing Istio on EKS (this might take a while)
namespace/istio-system created
configmap/istio-crd-10 created
configmap/istio-crd-11 created
configmap/istio-crd-12 created
serviceaccount/istio-init-service-account created
clusterrole.rbac.authorization.k8s.io/istio-init-istio-system unchanged
clusterrolebinding.rbac.authorization.k8s.io/istio-init-admin-role-binding-istio-system unchanged
job.batch/istio-init-crd-10 created
job.batch/istio-init-crd-11 created
job.batch/istio-init-crd-12 created
NAME                                         AGE
meshpolicy.authentication.istio.io/default   3m51s
[keptn|DEBUG] [2019-11-22 14:49:26] All custom resource definitions are available.
poddisruptionbudget.policy/istio-ingressgateway created
poddisruptionbudget.policy/istio-pilot created
poddisruptionbudget.policy/istio-sidecar-injector created
configmap/istio-security-custom-resources created
configmap/istio created
configmap/istio-sidecar-injector created
serviceaccount/istio-ingressgateway-service-account created
serviceaccount/istio-pilot-service-account created
serviceaccount/istio-cleanup-secrets-service-account created
clusterrole.rbac.authorization.k8s.io/istio-cleanup-secrets-istio-system unchanged
clusterrolebinding.rbac.authorization.k8s.io/istio-cleanup-secrets-istio-system unchanged
job.batch/istio-cleanup-secrets-1.3.1 created
serviceaccount/istio-security-post-install-account created
clusterrole.rbac.authorization.k8s.io/istio-security-post-install-istio-system unchanged
clusterrolebinding.rbac.authorization.k8s.io/istio-security-post-install-role-binding-istio-system unchanged
job.batch/istio-security-post-install-1.3.1 created
serviceaccount/istio-citadel-service-account created
serviceaccount/istio-sidecar-injector-service-account created
serviceaccount/istio-multi created
clusterrole.rbac.authorization.k8s.io/istio-pilot-istio-system unchanged
clusterrole.rbac.authorization.k8s.io/istio-citadel-istio-system unchanged
clusterrole.rbac.authorization.k8s.io/istio-sidecar-injector-istio-system unchanged
clusterrole.rbac.authorization.k8s.io/istio-reader unchanged
clusterrolebinding.rbac.authorization.k8s.io/istio-pilot-istio-system unchanged
clusterrolebinding.rbac.authorization.k8s.io/istio-citadel-istio-system unchanged
clusterrolebinding.rbac.authorization.k8s.io/istio-sidecar-injector-admin-role-binding-istio-system unchanged
clusterrolebinding.rbac.authorization.k8s.io/istio-multi unchanged
role.rbac.authorization.k8s.io/istio-ingressgateway-sds created
rolebinding.rbac.authorization.k8s.io/istio-ingressgateway-sds created
service/istio-ingressgateway created
service/istio-pilot created
service/istio-citadel created
service/istio-sidecar-injector created
deployment.apps/istio-ingressgateway created
deployment.apps/istio-pilot created
deployment.apps/istio-citadel created
deployment.apps/istio-sidecar-injector created
horizontalpodautoscaler.autoscaling/istio-ingressgateway created
horizontalpodautoscaler.autoscaling/istio-pilot created
mutatingwebhookconfiguration.admissionregistration.k8s.io/istio-sidecar-injector configured
Waiting for deployment "istio-ingressgateway" rollout to finish: 0 of 1 updated replicas are available...
deployment "istio-ingressgateway" successfully rolled out
[keptn|DEBUG] [2019-11-22 14:50:09] Deployment istio-ingressgateway in istio-system namespace available.
deployment "istio-pilot" successfully rolled out
[keptn|DEBUG] [2019-11-22 14:50:10] Deployment istio-pilot in istio-system namespace available.
deployment "istio-citadel" successfully rolled out
[keptn|DEBUG] [2019-11-22 14:50:10] Deployment istio-citadel in istio-system namespace available.
deployment "istio-sidecar-injector" successfully rolled out
[keptn|DEBUG] [2019-11-22 14:50:10] Deployment istio-sidecar-injector in istio-system namespace available.
NAME                                      READY   STATUS      RESTARTS   AGE
istio-citadel-6649cd766-rgxwd             1/1     Running     0          43s
istio-cleanup-secrets-1.3.1-n4487         0/1     Completed   0          44s
istio-ingressgateway-645d77db74-vqg8g     1/1     Running     0          43s
istio-init-crd-10-rlt5t                   0/1     Completed   0          45s
istio-init-crd-11-w68q9                   0/1     Completed   0          45s
istio-init-crd-12-kx6xv                   0/1     Completed   0          45s
istio-pilot-bf9fdf9fd-2zpmg               2/2     Running     0          43s
istio-security-post-install-1.3.1-ttgxw   0/1     Completed   0          44s
istio-sidecar-injector-5d6ff6d758-vspxm   1/1     Running     0          43s
[keptn|DEBUG] [2019-11-22 14:50:11] All pods are running in namespace istio-system.
[keptn|DEBUG] [2019-11-22 14:50:11] hostname of Istio Ingressgateway is available.
Can't load /root/.rnd into RNG
139922150061888:error:2406F079:random number generator:RAND_load_file:Cannot open file:../crypto/rand/randfile.c:88:Filename=/root/.rnd
Generating a RSA private key
.....................+++++
.+++++
writing new private key to 'key.pem'
-----
problems making Certificate Request
139922150061888:error:0D07A097:asn1 encoding routines:ASN1_mbstring_ncopy:string too long:../crypto/asn1/a_mbstr.c:107:maxsize=64
error: Cannot read file certificate.pem, open certificate.pem: no such file or directory
rm: cannot remove 'certificate.pem': No such file or directory
[keptn|ERROR] [2019-11-22 14:50:11] Installing Istio failed.
[keptn|ERROR] [2019-11-22 14:50:11] Stopping keptn installation. Already created resources are not deleted; run keptn uninstall to clean-up.
rpc error: code = Unknown desc = Error: No such container: 806a95f722bc0acfe635229404009853bf36611cf06ceb8a2d87d75ebe09604f%
```

**Comment by checkelmann** [comment]
Hi,

I was digging a bit more into the Problem.
So EKS or the Istio Installer will create a ELB with a generic name like `a47f87cd10e3722eab05d0a265b6f647-1924567589.eu-west-1.elb.amazonaws.com`.

setupIstio.sh queries the the Ingress DNS with `export DOMAIN=$(kubectl get svc istio-ingressgateway -o json -n istio-system | jq -r .status.loadBalancer.ingress[0].hostname)` which is retuning the right DNS but this DNS name is too long for the CN Part of the Certificate which is generated with openssl

`openssl req -nodes -newkey rsa:2048 -keyout key.pem -out certificate.pem  -x509 -days 365 -subj "/CN=$DOMAIN"`

This whole part was added within this commit https://github.com/keptn/keptn/commit/e837a720c36388430fa95c7f57a357cc040272f6#diff-2b25376623f303e1df6de62964c35391

**Comment by johannes-b** [comment]
Hello Christian, 

thanks for reporting this issue and providing the in-depth analysis! You are absolutely right with the improvement you suggest. 

I addressed this issue in: #1157

As soon as it is patched in the 0.6.0.beta release, I will let you know.
